src/advanced_fees_management.tf

```diff
@@ -0,0 +1,178 @@
+    SnapshotDebugger_EXTENSION_VERSION              = "disabled"
+    XDT_MicrosoftApplicationInsights_BaseExtensions = "disabled"
+    XDT_MicrosoftApplicationInsights_Mode           = "recommended"
+    XDT_MicrosoftApplicationInsights_PreemptSdk     = "disabled"
+    WEBSITE_HEALTHCHECK_MAXPINGFAILURES             = 10
+    TIMEOUT_DELAY                                   = 300
+    # Integration with private DNS (see more: https://docs.microsoft.com/en-us/answers/questions/85359/azure-app-service-unable-to-resolve-hostname-of-vi.html)
+    WEBSITE_ADD_SITENAME_BINDINGS_IN_APPHOST_CONFIG = "1"
+    WEBSITE_RUN_FROM_PACKAGE                        = "1"
+    WEBSITE_VNET_ROUTE_ALL                          = "1"
+    WEBSITE_DNS_SERVER                              = "168.63.129.16"
+
+    # Spring Environment
+    COSMOS_KEY1 = data.azurerm_key_vault_secret.afm_cosmos_key1.value
+    COSMOS_KEY2 = data.azurerm_key_vault_secret.afm_cosmos_key2.value
+    COSMOS_URI  = null
+
+    CORS_CONFIGURATION         = jsonencode(local.advanced_fees_management_cors_configuration)
+
+    WEBSITES_ENABLE_APP_SERVICE_STORAGE = false
+    WEBSITES_PORT                       = 8080
+
+    DOCKER_REGISTRY_SERVER_URL      = "https://${module.acr[0].login_server}"
+    DOCKER_REGISTRY_SERVER_USERNAME = module.acr[0].admin_username
+    DOCKER_REGISTRY_SERVER_PASSWORD = module.acr[0].admin_password
+  }
+
+  allowed_subnets = [module.apim_snet.id]
+  allowed_ips     = []
+
+  subnet_id = module.advanced_fees_management_snet[0].id
+
+  tags = var.tags
+}
+
+# afm cosmos configuration
+
+# cosmosdb account
+module "advanced_fees_management_cosmosdb_account" {
+  count  = var.advanced_fees_management_enabled ? 1 : 0
+
+  source   = "git::https://github.com/pagopa/azurerm.git//cosmosdb_account?ref=v2.1.18"
+  name     = format("%s-afm-cosmos", local.project)
+  location = var.location
+
+  resource_group_name = azurerm_resource_group.advanced_fees_management_rg.name
+  offer_type          = "Standard"
+  kind                = "GlobalDocumentDB"
+
+  public_network_access_enabled    = var.advanced_fees_management_cosmosdb_public_network_access_enabled
+  main_geo_location_zone_redundant = false
+
+  enable_free_tier          = false
+  enable_automatic_failover = true
+
+  consistency_policy = {
+    consistency_level       = "Strong"
+    max_interval_in_seconds = null
+    max_staleness_prefix    = null
+  }
+
+  main_geo_location_location = "westeurope"
+
+  # in order to disable redundancy in dev
+  main_geo_location_zone_redundant = var.advanced_fees_management_enabled
+
+  # for the PoC we are not interested to redundancy
+#  additional_geo_locations = [
+#    {
+#      location          = "northeurope"
+#      failover_priority = 1
+#      zone_redundant    = true
+#    }
+#  ]
+
+  # for the PoC we are not interested to backup
+  backup_continuous_enabled = false
+
+  is_virtual_network_filter_enabled = false
```

Comment:
it is preferable set to `true` and then add to `allowed_virtual_network_subnet_ids` whoever the access permissions